src/spec.c

```diff
@@ -1153,7 +1153,14 @@ void IndexSpec_FreeSync(IndexSpec *spec) {
 //---------------------------------------------------------------------------------------------
 
 void Indexes_Free(dict *d) {
-  arrayof(IndexSpec *) specs = array_new(IndexSpec *, 10);
+  // free the schema dictionary this way avoid iterating over it for each combination of
+  // spec<-->prefix
+  SchemaPrefixes_Free(ScemaPrefixes_g);
+  SchemaPrefixes_Create();
+  // cursor list is iterating through the list as well and consuming a lot of CPU
+  CursorList_Empty(&RSCursors);
+
+  arrayof(IndexSpec *) specs = array_new(IndexSpec *, dictSize(d));
```

Comment:
Wouldn't the spec will try to delete itself from `ScemaPrefixes_g` and from `RSCursors` structured?

Reply:
Yes. The same way it will try when removing a single spec. But since the list is empty it will be minimal effort